src/thing_creature.c

```diff
@@ -5364,6 +5369,481 @@ void illuminate_creature(struct Thing *creatng)
     lgt->radius <<= 1;    
 }
 
+void controlled_creature_pick_thing_up(struct Thing *creatng, struct Thing *picktng)
+{
+    if (picktng->class_id == TCls_Creature)
+    {
+        set_creature_being_dragged_by(picktng, creatng);
+    }
+    else
+    {
+        if ((picktng->owner != creatng->owner) )
+        {
+            if (thing_is_workshop_crate(picktng))
+            {
+                update_workshop_object_pickup_event(creatng, picktng);
+            }
+            else if ( (thing_is_spellbook(picktng)) || (thing_is_special_box(picktng)) )
+            {
+                update_library_object_pickup_event(creatng, picktng);
+            }
+        }
+        creature_drag_object(creatng, picktng);
+    }
+    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+    cctrl->pickup_object_id = picktng->index;
+    struct CreatureSound* crsound = get_creature_sound(creatng, CrSnd_Hurt);
+    unsigned short smpl_idx = crsound->index + 1;
+    thing_play_sample(creatng, smpl_idx, 90, 0, 3, 0, 2, FULL_LOUDNESS);
+    display_controlled_pick_up_thing_name(picktng, GUI_MESSAGES_DELAY);
+}
+
+void controlled_creature_drop_thing(struct Thing *creatng, struct Thing *droptng)
+{
+    creature_drop_dragged_object(creatng, droptng);
+    clear_messages_from_player(-81);
+    clear_messages_from_player(-86);
+    unsigned short smpl_idx, pitch;
```

Comment:
This line causes build warnings. I believe the compiles prefers it if you put it to 0.